### ISSUE TITLE
Make "interesting" metadata on DLPs to lead to Search 

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-archive#1884]
Nearly randomly I ended up on a DLP of a dandiset https://dandiarchive.org/dandiset/000629?search=000629&pos=1  and I wondered what other dandisets that author uploaded.  So I copy pasted name into our search and it worked nicely: https://dandiarchive.org/dandiset/search?search=%22Gillespie,%20Anna%22 . I got excited!

Then I thought - what other dandisets I could potentially find carrying the same datatype: https://dandiarchive.org/dandiset/search?search=ProcessingModule and more cool was -- "all" of the data types of this dandiset:
https://dandiarchive.org/dandiset/search?search=Position%20ElectricalSeries%20ProcessingModule%20SpatialSeries%20BehavioralEvents

![image](https://github.com/dandi/dandi-archive/assets/39889/a406fd52-472f-4da1-885f-b9a7c4de5711)

species?  https://dandiarchive.org/dandiset/search?search=Rattus%20norvegicus%20-%20Norway%20rat

So, even if our search might not be too fancy ATM, I think it is already a great tool and we should make use of it more!

What if for most of the metadata entries we add ability to turn them into search queries to look for similar/related content?  

Would it be easy to difficult to make it happen within current DLP frontend to augment some of those locations with `?` to follow every value or section name which would link to to the search url?

Later on we could improve search mechanism to support better language to limit to specific metadata fields .

**Comment by waxlamp** [comment]
> Would it be easy to difficult to make it happen within current DLP frontend to augment some of those locations with ? to follow every value or section name which would link to to the search url?

I'm not quite sure what you mean by this. Can you describe this more in terms of how you picture the UI for it? Is it something like a link or a button near metadata displays that says something like `[more like this]`?

**Comment by yarikoptic** [comment]
> Is it something like a link or a button near metadata displays that says something like [more like this]?

yes. besides it is more like `[more of this]` (of this author, modality, etc).

Basing on my initial description, to get 

![image](https://github.com/dandi/dandi-archive/assets/39889/21a5d187-af33-4d5e-af8d-3b1be8337c58)
into 
![image](https://github.com/dandi/dandi-archive/assets/39889/50e6487a-e2e9-4cbe-9874-336e5d726dbb)

where

![image](https://github.com/dandi/dandi-archive/assets/39889/e46a9a65-b232-40e2-8627-20a92b69e2c3)
 leads to https://dandiarchive.org/dandiset/search?search=%22Gillespie%2C+Anna%22 and gain it for all other similar items... so get smth like

![image](https://github.com/dandi/dandi-archive/assets/39889/a50cd01b-7ac9-42e6-8f2c-225b04c4b6ac)

or may be make it a default action for any of those when clicked , instead of adding dedicated magnifying glass.

**Comment by tekrajchhetri** [comment]
@yarikoptic I think [https://civicdb.org/welcome](https://civicdb.org/) is a good example for this.
